docs/content/preview/tutorials/AI/ai-ollama.md

```diff
@@ -242,10 +246,31 @@ app.get("/api/search", async (req, res) => {
 
 The `/api/search` endpoint specifies the embedding model and prompt to be sent to Ollama to generate a vector representation. This is done using the Ollama JavaScript library. The response is then used to execute a cosine similarity search against the dataset stored in YugabyteDB using pgvector. Latency is reduced by pre-filtering by news category, thus reducing the search space.
 
+```sql
+# backend/schema.sql
+CREATE EXTENSION IF NOT EXISTS vector;
+
+CREATE TABLE
+    news_stories (
+        id serial primary key,
+        link text,
+        headline text,
+        category varchar(50),
+        short_description text,
+        authors text,
+        date date,
+        embeddings vector (768)
+    );
+
+CREATE INDEX NONCONCURRENTLY ON news_stories USING ybhnsw (embeddings vector_cosine_ops);
+```
+
+The search speed is further increased by using [vector indexing](../../../explore/ysql-language-features/pg-extensions/extension-pgvector/#vector-indexing). YugabyteDB currently supports the HNSW (Hierarchical Navigable Small World) index type. This app uses cosine distance for indexing as the backend query is using cosine similarity search.
```

Comment:
The search speed is further increased by using [vector indexing](../../../explore/ysql-language-features/pg-extensions/extension-pgvector/#vector-indexing). YugabyteDB currently supports the Hierarchical Navigable Small World (HNSW) index type. This application uses cosine distance for indexing, as the backend query is using cosine similarity search.

This application is straightforward. Before executing similarity searches, embeddings for each news story must be generated and then stored in the database. Refer to the `generate_embeddings.js` script for details.